File: backends/go-beego/internal/domain/invitation/entity.go
package invitation

import "time"

type Invitation struct {
	ID        uint
	Token     string
	Role      int
	CreatedAt time.Time
	CreatedBy *uint
	UpdatedAt time.Time
	UpdatedBy *uint
	DeletedAt *time.Time
	DeletedBy *uint
	Version   int
}

File: backends/ruby-hanami/app/domain/invitation/entity.rb
# frozen_string_literal: true
#
# 招待エンティティを表すドメインオブジェクトモジュール。
#
# @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

module Domain
  module Invitation
    # 招待のエンティティです。
    Entity = Struct.new(
      :id, :token, :role,
      :created_at, :created_by, :updated_at, :updated_by,
      :deleted_at, :deleted_by, :version,
      keyword_init: true
    )
  end
end
